From f4f98eb5cc9121de1b59f55c334d260e015ea37b Mon Sep 17 00:00:00 2001
From: Steven Xu <stevenxxiu@gmail.com>
Date: Tue, 2 Mar 2021 00:03:17 +1100
Subject: feat: use `g_utf8_collate_key()` instead of
 `g_utf8_collate_key_for_filename()`, so files beginning with `_` is sorted
 before files beginning with numbers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

- [Files not properly sorted by name 路 Issue #2135 路 linuxmint/nemo](https://github.com/linuxmint/nemo/issues/2135#issuecomment-492007809)

This still requires a custom locale that sorts `_` before numbers.
---
 libnemo-private/nemo-file.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/libnemo-private/nemo-file.c b/libnemo-private/nemo-file.c
index eb23767..ea27bfa 100644
--- a/libnemo-private/nemo-file.c
+++ b/libnemo-private/nemo-file.c
@@ -261,7 +261,12 @@ nemo_file_set_display_name (NemoFile *file,
 		}
 
 		g_free (file->details->display_name_collation_key);
-		file->details->display_name_collation_key = g_utf8_collate_key_for_filename (display_name, -1);
+		// Use `g_utf8_collate_key()` instead of `g_utf8_collate_key_for_filename()`, so files beginning with `_` is
+		// sorted before files beginning with numbers, cf [Files not properly sorted by name 路 Issue #2135 路 linuxmint/nemo]
+		// (https://github.com/linuxmint/nemo/issues/2135#issuecomment-492007809)
+		//
+		// This still requires a custom locale that sorts `_` before numbers.
+		file->details->display_name_collation_key = g_utf8_collate_key (display_name, -1);
 	}
 
 	if (g_strcmp0 (eel_ref_str_peek (file->details->edit_name), edit_name) != 0) {
-- 
2.30.2

